openedx/core/djangoapps/credit/utils.py

```diff
@@ -0,0 +1,35 @@
+"""
+Utilities for the credit app.
+"""
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+
+def get_course_blocks(course_key, category):
+    """
+    Retrieve all XBlocks in the course for a particular category.
+
+    Returns only XBlocks that are published and haven't been deleted.
+    """
+    return [
+        block for block in modulestore().get_items(
+            course_key,
+            qualifiers={"category": category},
+            revision=ModuleStoreEnum.RevisionOption.published_only,
+        )
+        if _is_in_course_tree(block)
+    ]
+
+
+def _is_in_course_tree(block):
+    """
+    Check that the XBlock is in the course tree.
+
+    It's possible that the XBlock is not in the course tree
+    if its parent has been deleted and is now an orphan.
+    """
+    ancestor = block.get_parent()
+    while ancestor is not None and ancestor.location.category != "course":
+        ancestor = ancestor.get_parent()
+
+    return ancestor is not None
```

Comment:
This hasn't been merged in yet (and won't be for a while) so feel free to copy but I think it makes sense to have only one function in the code base for determining if there is a path to a block:
https://github.com/edx/edx-platform/commit/0da1a577fc081197d5c1dbdb84b7caeaaf804c9b#diff-8b10585ad573b7a3cf514f68d7976a83L9